packages/nimble-components/src/table/specs/table-columns-hld.md

```diff
@@ -85,7 +85,21 @@ This interface could possibly be expanded in the future to communicate relevant
 This abstract class is what a column web component (i.e. a slotted column element) must extend.
 
 ```TS
-abstract class TableColumn<TCellRecord extends TableRecord = TableRecord, TColumnConfig = {}> {
+abstract class TableColumn<TCellData extends TableRecord = TableRecord, TColumnConfig = {}> {
+    // An optional ID to associated with the column.
+    @attr({ attribute: 'column-id' })
+    columnId?: string;
+
+    // The name of the slot containing the action menu for this column, or `undefined` to indicate
+    // that the column does not have an action menu.
+    // Note: Multiple columns can specify the same slot.
+    @attr({ attribute: 'action-menu'})
+    actionMenu?: string;
```

Comment:
Double checking: do authors of custom columns need to do anything to support that column having an action menu? My understanding is that they don't. And I guess this is reflected by these attributes not being `abstract`. But just want to make sure I'm following correctly.

Reply:
Your understanding is correct. My guess is that we'll have quite a few additional attributes that behave this way -- whether the column is sortable/resizable/groupable, the size configuration of the column, etc.